eclair-core/src/main/scala/fr/acinq/eclair/wire/CommonCodecs.scala

```diff
@@ -84,14 +85,18 @@ object CommonCodecs {
         case Attempt.Successful(b) =>
           b.value match {
             case 0xff => verifyMinimalEncoding(0x100000000L, uint64L.decode(b.remainder))
-            case 0xfe => verifyMinimalEncoding(0x10000L, uint32L.decode(b.remainder))
-            case 0xfd => verifyMinimalEncoding(0xfdL, uint16L.decode(b.remainder).map(b => b.map(_.toLong)))
-            case _ => Attempt.Successful(DecodeResult(b.value.toLong, b.remainder))
+            case 0xfe => verifyMinimalEncoding(0x10000L, uint32L.decode(b.remainder).map(b => b.map(UInt64(_))))
+            case 0xfd => verifyMinimalEncoding(0xfdL, uint16L.decode(b.remainder).map(b => b.map(UInt64(_))))
+            case _ => Attempt.Successful(DecodeResult(UInt64(b.value), b.remainder))
           }
         case Attempt.Failure(err) => Attempt.Failure(err)
       }
     })
 
+  // This codec can be safely used for values < 2^63 and will fail otherwise.
+  // It is useful in combination with variableSizeBytesLong to encode/decode TLV lengths because those will always be < 2^63.
+  val varlong: Codec[Long] = varint.narrow(l => if (l <= UInt64(Long.MaxValue)) Attempt.successful(l.toBigInt.toLong) else Attempt.failure(Err(s"overflow for value $l")), l => UInt64(l))
```

Comment:
This one need to be a `Codec[Long]` because there is no safe conversion from `UInt64` to `Long` (since `UInt64` can overflow a `Long`) and I need to use that in a `variableSizeBytesLong` codec for TLV (and `variableSizeBytesLong` requires a `Codec[Long]`).

Reply:
But if there's some type magic that allows me to directly use the `varint` codec in `variableSizeBytesLong` even though it's a `Codec[UInt64]` and not a `Codec[Long]`, I'm interested!

Reply:
```suggestion
  val varintoverflow: Codec[Long] = varint.narrow(l => if (l <= UInt64(Long.MaxValue)) Attempt.successful(l.toBigInt.toLong) else Attempt.failure(Err(s"overflow for value $l")), l => UInt64(l))
```

Ok if you rename it to `varintoverflow` it makes more sense and is consistent with the `unit64`/`unit64overflow`.

Reply:
Good point, done.